### DISCUSSION TITLE
Aiuto creazione Entità e Statistiche

### DISCUSSION THREAD

**Discussion by matteo289** [indomus/forum#2858] [HUB: Home Assistant]
Ciao a tutti, ho un problema nella creazione di entità e statistiche su ha. Ho copiato un codice da un package, ho modificato i nomi delle entità come vorrei averle, ma non mi vengono create. Qualcuno può aiutarmi?

```
sensor:
  - sensor:
      state: "{% set hours = states('sensor.valvola_sala_giorno') | float %}
        {% set minutes = ((hours % 1) * 60) | int %}
        {% set hours = (hours - (hours % 1)) | int %}
        {{ '%02i:%02i'%(hours, minutes) }} H"

  - sensor:
      state: "{% set hours = states('sensor.valvola_sala_mese') | float %}
        {% set minutes = ((hours % 1) * 60) | int %}
        {% set hours = (hours - (hours % 1)) | int %}
        {{ '%02i:%02i'%(hours, minutes) }} H"

  - sensor:
      state: "{% set hours = states('sensor.valvola_sala_anno') | float %}
        {% set minutes = ((hours % 1) * 60) | int %}
        {% set hours = (hours - (hours % 1)) | int %}
        {{ '%02i:%02i'%(hours, minutes) }} H"

  - sensor:
      - name: "Stato Valvola Sala"
        state: >-
          {% if is_state('climate.sala', 'heat') %}
          Boost
          {% elif is_state('climate.sala', 'auto') %}
          Automatico
          {% elif is_state('climate.sala', 'off') %}
          Spenta
          {% endif %}
```

1) stato valvola sala --> non mi crea l'entità
2) sensor.valvola_sala_mese/giorno/anno  --> non si crea l'entità

queste variabili dovrei usarle per una platform history_stats

Grazie

**Comment by Scialla** [comment]
Hai sbagliato tutto....
non puoi  mettere `sensor:` e sotto di lui `- sensor:`
hai estrapolato alcune parti di codice dal proprio contesto
devi creare dei `template` sensor per i quali non hai messo parti fondamentali, manca pure il nome!

Dai un occhio a questa guida per capire come si fà:
https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-1-creare-dei-sensori-template/

aiutati con gli strumenti sviluppatore nel tab "modelli"

**Comment by matteo289** [comment]
io ho capitato il codice sperando andasse bene ...
quindi alla fine devo riscrivere tutto giusto?

**Reply by Scialla** [reply]
si, impara come si fanno i sensori template leggendo la guida, poi potrai crearti quelli che ti servono. Non basta fare copia e incolla, devi capire "come funzionano" i sensori che ti interessano in modo da crearteli

PS non creare nuovi thread quando rispondi, fai il reply sotto l'ultima risposta

**Reply by matteo289** [reply]
ok Scusami....

Allora ti chiedo per ora solo questo. Il codice scritto cosi, come ho guardato dalla guida, 


```
template:
  - sensor:
      stato_valvola_sala:
        friendly_name: "Stato valvola Sala"
        value_template: >-
          {% if is_state('climate.sala', 'heat') %}
          Boost
          {% if is_state('climate.sala', 'auto') %}
          Automatico
          {% if is_state('climate.sala', 'off') %}
          Spento
          {% endif %}
```


è corretto? perchè a me il sensore non lo crea. e non ho altri codici scritti se non input_number e utility meter

**Reply by matteo289** [reply]
in questa configurazione

### Sensor ###
template:
  - sensor:
      - name: Stato Valvola Sala
        friendly_name: "Stato valvola Sala"
        value_template: >-
          {% if is_state('climate.sala', 'heat') %}
          Boost
          {% if is_state('climate.sala', 'auto') %}
          Automatico
          {% if is_state('climate.sala', 'off') %}
          Spento
          {% endif %}

  - sensor:
      - name: Temperatura Sala
        state: "{{ states.climate.sala.attributes.current_temperature }}°C"

Temperatura Sala è creata
Stato Valvola Sala non si crea

come mai?
